### PR TITLE
Fix linebreak fallback tofu

### LINKED ISSUE #2105: #sym.arrow.r.squiggly after \ will produce a tofu symbol
### Description

Enter this in a.typ:

```typst
a. Form #sym.arrow.r.squiggly (Form #sym.subset Form) \
#sym.arrow.r.squiggly (aaa)
```

will give

image

**Expected behavior**

Get from typst.app

image
**Package/Software version:**
Typst: nightly version https://github.com/typst/typst#073effc7
OS: Arch Linux x86_64

I cannot reproduce it on the typst.app.

### Reproduction URL

_No response_

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

### PR DESCRIPTION
**Pull request by Enter-tainer**
fix #2105.

When linebreak is present, the text sent to `shape_segments` is `\nACTUAL_TEXT`. And we are using the first char to select fallback fonts. So that's the problem

### CODE REVIEW

File: crates/typst/src/text/font/book.rs [hunk 1]
```diff
@@ -93,8 +93,8 @@ impl FontBook {
         variant: FontVariant,
         text: &str,
     ) -> Option<usize> {
-        // Find the fonts that contain the text's first char ...
-        let c = text.chars().next()?;
+        // Find the fonts that contain the text's first non-space char ...
+        let c = text.chars().find(|c| !c.is_whitespace())?;
```
Comment (PgBiel) [inline crates/typst/src/text/font/book.rs:97]: What happens if all characters are whitespace? I believe that'd receive the same treatment as an empty string, which shouldn't be that bad, right?

Comment (Enter-tainer) [reply crates/typst/src/text/font/book.rs:97]: when the whole string is space, it wont even be shaped. there is a `if` handling this

Comment (PgBiel) [reply crates/typst/src/text/font/book.rs:97]: Got it, thanks!

**Comment by laurmaedje** [comment]
Makes sense, thanks!

**Pull request merged**